decoder/context.go

```diff
@@ -16,8 +21,79 @@ type DecoderContext struct {
 	// CodeLenses represents a slice of executable lenses
 	// which will be executed in the exact order they're declared
 	CodeLenses []lang.CodeLensFunc
+
+	CompletionHooks        CompletionFuncMap
+	CompletionResolveHooks CompletionResolveFuncMap
```

Comment:
Can you just briefly explain in comments above each field how these work? i.e. that this is place to register hooks, which makes them available for the schema and based on the schemas, they get looked up during `CompletionAtPos()` and `ResolveCandidate()` based on the map key - or something along these lines.